pkg/operator/ceph/pool/status.go

```diff
@@ -46,12 +44,24 @@ func updateStatus(ctx context.Context, client client.Client, poolName types.Name
 		pool.Status = &cephv1.CephBlockPoolStatus{}
 	}
 
+	// add pool ID to the status
+	if status == cephv1.ConditionReady && pool.Status.PoolID == 0 {
```

Comment:
only doing the ceph query for pool details once - when pool is ready and poolID is not set already. 
That should be ok. I don't see pool ID changing.  Also, don't think pool ID can be `0` so its ok to use `status.PoolID` as int.